### ISSUE TITLE
Implement RFC 10 (master->main, etc.)

### ISSUE THREAD

**Issue by brianlheim** [supercollider/supercollider#5251]
RFC 10 (https://github.com/supercollider/rfcs/pull/10) "Replace oppressive terminology with more accurate alternatives" was merged this summer; we held off implementing it during the releases to keep things simple, but it should really be done now that we have a break between the 3.11.2 and 3.12.0 release.

See RFC text for more details including implementation plan.

Would anyone like to help with this?

**Comment by muellmusik** [comment]
Hey Brian, I’m happy to help. Do you have a plan to split it up?

**Comment by brianlheim** [comment]
@muellmusik thanks! yes, there are 5 major parts as i see it:
1. announce that these changes are coming and what action, if any, is required
2. change all 'master' branches to 'main' (via github UI)
3. create PRs to update language as described here: https://github.com/supercollider/rfcs/pull/10/files#diff-37ceb2dcb38ee886e4607ad26efb1c988afeb9be76798026cbf40eb4b1517db5R34-R47
4. update wikis and any other documents not in source control
5. update open PRs that point to `master` to point to `main` instead

(1) and (3) can start happening right now. for most people, i don't think this should require any workflow change other than checking out a different branch. however, i do want to pay special attention to whether this would affect submodules (because it seems like they're always the weakest link >_> ). someone should simulate this change on their own forks and see whether it causes any issues in updating submodules. right now, i can't think of a reason why it would, since checkouts are based on commit hashes and not branch names, but i have low confidence about that.

**Comment by muellmusik** [comment]
Cool I'll try to make a start on 3 in the coming week or so.

**Comment by brianlheim** [comment]
@muellmusik any update on this?

**Comment by muellmusik** [comment]
Hey Brian,

Just been swamped with marking and term prep but will push something soon.

**Comment by brianlheim** [comment]
I tried out renaming master->main in some test repositories and it went pretty smoothly, so i'd like to finally get around to implementing this. here's a draft of the message i'll be sending on sc-users, sc-dev, and scsynth.org. unless anyone has concerns, i'll send this out on April 5.

----

[subject] Notice: Renaming master to main on April 17

Hey all,

As part of implementing SuperCollider RFC #10 (https://github.com/supercollider/rfcs/pull/10), the dev team will soon rename "master" branches in all SuperCollider organization repositories will to "main". This change will happen on Saturday, April 15, and will affect all 21 repositories listed here: https://github.com/supercollider. This does not affect the https://github.com/supercollider-quarks organization, which requires extra discussion due to the design of Quarks.

On that day, we will use the GitHub UI to rename the "master" branch of each repository to "main" as indicated here: https://github.com/github/renaming.

Another message will follow this one once the change is complete, at which point action will be needed on your part. *Do not perform any of these steps until after you've seen that message!*

If you have forked any of these repositories on GitHub, you will be able to update by performing the same renaming operation on your own repository via the GitHub UI: go to your repo, click "branches", click the pencil icon next to the "master" branch to rename, and set the name to "main".

If you have cloned any of these repositories locally, you will be able to follow these steps to update your branch. Note that the GitHub UI also shows a reminder with similar instructions.

    # "move" master to main
    git branch -m master main

    # set the correct upstream -- this will work regardless of whether you had the
    # SuperCollider remote named as "upstream" or "origin"
    git branch -u $(git config branch.main.remote)/main main

    # update your local reference for the SuperCollider remote
    git remote set-head $(git config branch.main.remote) -a

This has no impact on repositories which are git submodules under ordinary usage. You will need to repeat the steps for updating a local clone if you are developing a submodule locally.

Any questions should be directed to the GitHub issue thread: https://github.com/supercollider/supercollider/issues/5251. We'll do our best to update if there are any further changes necessary for forks and clones.

Regards,
Brian

**Comment by alln4tural** [comment]
There's an extra "will" in the first sentence:
.. the dev team will soon rename "master" branches in all SuperCollider organization repositories will to "main".

Cheers,
eddi

**Comment by brianlheim** [comment]
I've updated all the branch names now, and opened 5 PRs (see references above). What needs to happen next:
- all 5 PRs reviewed and merged
- merge new PR for SC to update the hidapi and scvim submodules in develop
- announcement

and then we can close this.

**Comment by brianlheim** [comment]
thanks for the quick reviews @dvzrv !
